include/packetgraph/thread.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2017 Outscale SAS
+ *
+ * This file is part of Butterfly.
+ *
+ * Butterfly is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published
+ * by the Free Software Foundation.
+ *
+ * Butterfly is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Butterfly.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _PG_THREAD_H
+#define _PG_THREAD_H
+
+/* TODO: error handeling ? */
+
+#include <packetgraph/errors.h>
+#include <packetgraph/graph.h>
+
+enum pg_thread_state {
+	PG_CANT_RECIVE_STATE = -1,
+	PG_THREAD_STOPPED,
+	PG_THREAD_BROKEN,
+	PG_THREAD_RUNNING
+};
+
+#define pg_thread_states_to_str(state)		\
+	(pg_thread_str_states[state + 1])
+extern const char *pg_thread_str_states[PG_THREAD_RUNNING + 2];
+
+int16_t pg_thread_max(void);
+
+int16_t pg_thread_init(struct pg_error **errp);
+
+int pg_thread_set_core(int16_t thread_id, int core_mask);
+
+int pg_thread_set_priority(int16_t thread_id, int priority);
+
+int pg_thread_run(int16_t thread_id);
+
+int pg_thread_stop(int16_t thread_id);
+
+enum pg_thread_state pg_thread_state(int16_t thread_id);
+
+/**
+* @graph_id set on which graph the error occur
+* @errp error
+* @return the number of error left, 0 if last error or -1 if no error pop
+*/
```

Comment:
Need better doc formating: title + description + doxygen indentation for parameter description